eng/pipelines/templates/stages/archetype-sdk-tests.yml

```diff
@@ -114,6 +114,8 @@ stages:
   - ${{ if or(contains(parameters.Clouds, cloud.key), and(contains(variables['Build.DefinitionName'], 'tests-weekly'), contains(parameters.SupportedClouds, cloud.key))) }}:
     - ${{ if not(contains(parameters.UnsupportedClouds, cloud.key)) }}:
       - stage: ${{ cloud.key }}_${{ parameters.TestName }}
+        variables:
```

Comment:
Why do we need to add this variable? Since this yml file is shared between all the SDKs,  can you ran another SDK's pipeline to see if it will cause any regression?

Reply:
Introducing a variable no one else is using feels pretty safe to me. :-) 

Sure, let me try one just to be sure.

Reply:
Why we need this new variable?

Reply:
I see that we only publish coverage only in Public cloud.

Reply:
This value should already be available via `$(System.StageName)`.

Reply:
I'm open to a more clearly named variable for this purpose, but to @minnieliu's point, it will need to be done in a way that is common to not only all pipelines, but all languages. That would likely mean doing it here: https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml#L92

It's a little tricky because of how you're trying to reference something at this top level to be called lazily, meaning we can't reference `${{ parameters.CloudConfig.Cloud }}` which would be accessible at the job yaml definition scope.

My thinking right now is let's keep it simple with stage name, since I expect to remove the need for this coverage definition as we will eventually add generic coverage opt-in for both live and ci pipelines.

Reply:
Honestly, I think archetype-sdk-tests-generate.yml may not be the best place since each sdk may still overwrite UnsupportedClouds in its own tests.yml (with archetype-sdk-tests.yml as template) unless you move the top level iteration in archetype-sdk-tests.yml to higher level. To me, the closest place to overwrite the cloud value (regardless which variable holds the value) is right within the iteration over the clouds (currently in archetype-sdk-tests.yml).

Reply:
> This value should already be available via `$(System.StageName)`.

I can give that a try

Reply:
Actually, I don't see $(System.StageName) being set in archetype-sdk-tests.yml. If it take value from the "stage" name, more than likely it'll read like "Public_LiveTest"

Reply:
Ah, yep didn't catch that Java was appending some strings to the stage name. You could try the more flexible `contains($(System.StageName), 'Public')` or `startsWith(...)` if you didn't want to hardcode the whole thing.

Reply:
I wasn't suggesting to define the cloud config, etc. in the generate file, but rather to set the variable there, e.g.

```
pwsh: |
  Write-Host "##vso[task.setvariable variable=CloudName;]${{ parameters.CloudConfig.Cloud }}"
```

Reply:
All this said, I think similar to the comments in the .net PR, we should expect to make this more generic and not have to pass in these commands from the tests.yml file.

Reply:
I think generalization is a good idea. But modifying engineering system, especially code under eng\pipelines, is not our expertise. It can be done after this works for us. :-)

Reply:
While I'm not necessarily opposed to adding a new parameter here I do think we old do what @benbp suggested for you current usage (i.e. `startsWith($(System.StageName), 'Public')`)